docs/proposals/mux-logging-service.md

```diff
@@ -0,0 +1,305 @@
+* `openshift_logging_use_mux_service` - `True`/`False` - default `False` - if
+  `True`, expose the mux service external to the cluster, and configure the mux
+  fluentd to accept logs from outside the cluster
+* `openshift_logging_mux_hostname` - default
+  "mux."`openshift_master_default_subdomain`.  This is the hostname that
+  clients outside the cluster will use, the one used in the mux TLS server cert
+* `openshift_logging_mux_port` - default 24284 - the port used by the mux Service
+  to listen for `secure_forward` protocol log messages
+
+Other parameters are similar to parameters for the other components, for
+example `openshift_logging_mux_cpu_limit`,
+`openshift_logging_mux_memory_limit`, `openshift_logging_mux_replica_count`,
+`openshift_logging_mux_nodeselector`
+
+Sending Records to mux
+-------------------------------
+
+### Fluentd secure_forward ###
+
+You will need the mux CA cert and shared_key.  You can obtain them from the
+logging cluster like this:
+
+    # oc login --username=system:admin
+    # oc project logging
+    # oc get secret logging-mux --template='{{index .data "ca"}}' | base64 -d > mux-ca.crt
+    # oc get secret logging-mux --template='{{index .data "shared_key"}}' | \
+      base64 -d > mux-shared-key
+
+The client side setup should look something like this:
+
+    <match **something**>
+      @type secure_forward
+      self_hostname forwarding-${hostname}
+      ca_cert_path /path/to/mux-ca.crt
+      secure yes
+      enable_strict_verification true
+      shared_key "#{File.open('/path/to/mux-shared-key') do |f| f.readline end.rstrip}"
+      <server>
+        host mux.logging.test
+        hostlabel mux.logging.test
+        port 24284
+      </server>
+    </match>
+
+Fluentd collectors running in the cluster will mount the logging-mux secret at
+`/etc/fluent/muxkeys` in order to use the `ca` and `shared_key`.
+
+### Tag ###
+
+The log records coming from the client should be tagged in the following
+format: `project.$namespace_name.other.data`
+* `project` is the literal string "project"
+* `$namespace_name` is the name of an OpenShift namespace.  For example,
+  something returned by the command `oc get projects`.  With OpenShift 1.5/3.5
+  and later, there is a 63 character limit on the namespace name.
+* `other.data` - mux will ignore the rest of the tag parts after
+  `$namespace_name` - this can be omitted - you can use a tag like
+  `project.namespacename`
+
+These logs will be written to Elasticsearch in an index with a name like the
+following: `project.$namespace_name.$namespace_uuid.YYYY.MM.DD`
+* `project` is the literal string "project"
+* `$namespace_name` is the name of the OpenShift namespace
+* `$namespace_uuid` is the UUID of the OpenShift namespace - fluentd will look
+  this up from OpenShift/Kubernetes using the kubernetes metadata filter
+* `YYYY.MM.DD` is the year, month, and day
+
+The namespaces used in the tag must be added ahead of time to OpenShift - for
```

Comment:
Is there a scenario where this is not already the case?  You can not run a container without it being part of a project which must exist ahead of time.